src/attribution/onCLS.ts

```diff
@@ -42,11 +42,12 @@ const attributeCLS = (metric: CLSMetric): void => {
           largestShiftEntry: largestEntry,
           loadState: getLoadState(largestEntry.startTime),
         };
+        return;
       }
     }
-  } else {
-    (metric as CLSMetricWithAttribution).attribution = {};
   }
+  // Set an empty object if no other attribution has been set.
+  (metric as CLSMetricWithAttribution).attribution = {};
```

Comment:
This works, but took me a while to grok.

It's not so much "if no other attribution has been set" as, "if not already returned previously with an attribution".

Alternatives:
1. Initialise the empty attribution at the top, and override it where necessary (rather than set it at the end... sometimes).
2. Set it with something like this:

```js
(metric as CLSMetricWithAttribution).attribution = (metric as CLSMetricWithAttribution).attribution || {};
```

Think either of those would be clearer? Or think it's clear enough to stick with what you have?

Reply:
I agree that initializing it at the top is a bit easier to read. However, I didn't really want to do that because I don't like that idea of creating an empty object just to not use it (which will be the case 99% of the time).

This may be a micro-optimization, but given that this is third-party code I thought it would be better to optimize for performance rather than clarify. If this were my own code I'd probably do it the way you suggest.

Reply:
Fair point. It actually looks easier in the code - just took me a moment when looking at the git diff to get what you were doing.